Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -1818,11 +1818,11 @@ - (void)scrollViewWillBeginDragging:(UIScrollView *)scrollView
 
 #if ENABLE(ASYNC_SCROLLING)
     // FIXME: We will want to detect whether snapping will occur before beginning to drag. See WebPageProxy::didCommitLayerTree.
-    auto* coordinator = downcast<WebKit::RemoteScrollingCoordinatorProxyIOS>(_page->scrollingCoordinatorProxy());
-    ASSERT(scrollView == _scrollView.get());
-    [_scrollView _setDecelerationRateInternal:(coordinator && coordinator->shouldSetScrollViewDecelerationRateFast()) ? UIScrollViewDecelerationRateFast : UIScrollViewDecelerationRateNormal];
-
-    coordinator->setRootNodeIsInUserScroll(true);
+    if (auto* coordinator = downcast<WebKit::RemoteScrollingCoordinatorProxyIOS>(_page->scrollingCoordinatorProxy())) {
+        ASSERT(scrollView == _scrollView.get());
```

Comment:
This debug assertion probably doesn't need to be behind the if statement.

Reply:
Hmm yes I agree, so I've moved it out in https://github.com/WebKit/WebKit/commit/df06269c9833e989801086e5b4d008ab5e966d5a